### ISSUE TITLE
Add test for successful_ignitions mode for FATES-SPITFIRE

### ISSUE THREAD

**Issue by glemieux** [ESCOMP/CTSM#1266]
Per @ekluzek [comment](https://github.com/ESCOMP/CTSM/pull/1264#discussion_r566406680), we should create a testmod option for enabling `fates_spitfire_mode = 3`.  Currently this is only referenced in `Fates_nat_and_anthro_ignitions` testmod (which itself enables `fates_spitfire_mode = 4`):

https://github.com/ESCOMP/CTSM/blob/5f348cac1919e70df19a6edbbf7b970c3877ce22/cime_config/testdefs/testmods_dirs/clm/Fates_nat_and_anthro_ignitions/README#L14-L26

**Comment by glemieux** [comment]
@ckoven do we need permission to move this data set into the CESM inputdata file structure?

**Comment by ckoven** [comment]
possibly? I'll ask.

**Comment by wwieder** [comment]
@ekluzek and @billsacks what do we need to do to bring this into  CESM inputdata?

**Comment by ekluzek** [comment]
> @ekluzek and @billsacks what do we need to do to bring this into CESM inputdata?

It's an easy task to do. We just need to copy it over into $DIN_LOC_ROOT/lnd/clm2/firedata and then import it into svn inputdata. We also need to setup a test configuration for this. This dataset is only for California, so we should have a surface dataset that's regional and just covers California or a part of California. The hard part of this is Charlie's ask for permission  to save the dataset that he mentions doing above.

**Comment by glemieux** [comment]
Just a heads up that I've got a branch in the works to address this issue and some of the other fates-related test issues (#1265,
#1273).

**Comment by lmkueppers** [comment]
I'll check with Yufang and Bin about the dataset. I presume that including means that it would be publicly available to anyone, right?

**Comment by ekluzek** [comment]
> I'll check with Yufang and Bin about the dataset. I presume that including means that it would be publicly available to anyone, right?

Yes, exactly. It would be publicly available for anyone. It wouldn't be obvious to find it, as it would be "hidden" in all the myriad of CESM input files. 

As an alternative to publicly available to anyone, we could reference to its location on cheyenne, and only use it there. I'd prefer if it could just be part of the normal CESM system though so would be downloaded if anyone happened to do this test. It wouldn't normally come up for anything standard it would require someone to have to do this specific test.

**Comment by ekluzek** [comment]
Currently the file doesn't have general meta-data as to the source of the data and who put it together. And for example to any papers regarding it. I'd like to have that kind of meta-data on the file as part of putting it into inputdata. So the credit to the authors of the data and the manipulations they did to get it in this format could be part of this exercise. And as a matter of fact I'd prefer it and would like to get all that kind of information. So we would give credit to Yufang and Bin about their work on this. @lmkueppers thanks for looking into this -- appreciate it.

**Comment by glemieux** [comment]
> Currently the file doesn't have general meta-data as to the source of the data and who put it together. And for example to any papers regarding it. I'd like to have that kind of meta-data on the file as part of putting it into inputdata. 

@ekluzek is there a reference guide for what meta-data is required for the inputdata process?  Is this something that I or Ryan can help process?

**Comment by ekluzek** [comment]
> @ekluzek is there a reference guide for what meta-data is required for the inputdata process? Is this something that I or Ryan can help process?

@glemieux talk offline. To add notes to what we talked about. Here are some documents

One document for CESM data is here. It's main point is to use the CF conventions.

https://www.cesm.ucar.edu/management/docs/data.mgt.plan.2011.pdf

CF conventions are documented here:

https://cfconventions.org/Data/cf-conventions/cf-conventions-1.8/cf-conventions.html

We are not perfect in our adoption of CF conventions and don't tend to use standard names. But, it's a good goal to strive for. Some good global attributes in the CF convention to add to files:

Conventions, comment, history, institution, references, source, title

Enough information that you'd have some hope of replicating the data is good. Since, data is used in scientific simulations you also want to have a way to document it. And you want to give credits to the originators of the data.

**Comment by ekluzek** [comment]
ncdump of the dataset shows the following. Which shows long_name, units, and mode should be added to the variable "ign", as well as global attributes to describe the data and it's source and manipulations done on it. time_bounds should be added (the start and ending time of the observed interval that it's averaged over (assuming the data is averaged over the interval). And "cell_methods" should be added to observed variables with the time dimension. We aren't consistent on this but the CF convention has Coordinates added to ancillary variables to point to the list of Coordinate variables for that variable. The coordinate variables here are lat, lon, and time.

netcdf ignition_1980_to_2016_monthly_20190801 {
dimensions:
	lon = 131 ;
	lat = 122 ;
	time = 444 ;
variables:
	double lat(lat) ;
		lat:long_name = "latitude" ;
		lat:units = "degrees_north" ;
		lat:mode = "time-invariant" ;
	double lon(lon) ;
		lon:long_name = "longitude" ;
		lon:units = "degrees_east" ;
		lon:mode = "time-invariant" ;
	int ign(time, lon, lat) ;
	double EDGEE ;
		EDGEE:long_name = "eastern edge in the data" ;
		EDGEE:units = "degrees_east" ;
		EDGEE:mode = "time-invariant" ;
	double EDGEN ;
		EDGEN:long_name = "northern edge in the data" ;
		EDGEN:units = "degrees_north" ;
		EDGEN:mode = "time-invariant" ;
	double EDGES ;
		EDGES:long_name = "southern edge in the data" ;
		EDGES:units = "degrees_north" ;
		EDGES:mode = "time-invariant" ;
	double EDGEW ;
		EDGEW:long_name = "western edge in the data" ;
		EDGEW:units = "degrees_east" ;
		EDGEW:mode = "time-invariant" ;
	double LATIXY(lat, lon) ;
		LATIXY:long_name = "latitude" ;
		LATIXY:units = "degrees_north" ;
		LATIXY:mode = "time-invariant" ;
	double LONGXY(lat, lon) ;
		LONGXY:long_name = "longitude" ;
		LONGXY:units = "degrees_east" ;
		LONGXY:mode = "time-invariant" ;
	double time(time) ;
		time:long_name = "observation time" ;
		time:units = "days since 1980-01-01 00:00:00" ;
		time:calendar = "noleap" ;
	double lnfm(time, lat, lon) ;
		lnfm:long_name = "number of ignitions" ;
		lnfm:units = "number/km2/hr" ;
		lnfm:mode = "time-dependent" ;
}

**Comment by glemieux** [comment]
Reviewing the fates-side code for this mode, I'd like to reassess if we actually need to use a custom successful ignitions dateset.  As noted  in the `fates_spitfire_mode = 4` test `README`: https://github.com/ESCOMP/CTSM/blob/e1f563b8345400d33421263f05b237f13e6b0e3d/cime_config/testdefs/testmods_dirs/clm/FatesFireLightningPopDens/README#L6-L12

the fates-side of the code sets the `cloud_to_ground_strikes` modifer to the incoming lightning to one, meaning that all strikes will result in an ignition as seen here:
https://github.com/NGEET/fates/blob/37c28ce7e94751905542cb29c096c74cc250d7eb/fire/SFMainMod.F90#L246-L251

That said, I'm naive on the nature of the default lightning dataset set by clm.  @ekluzek, @slevis-lmwg , @adrifoster thoughts?

**Comment by slevis-lmwg** [comment]
@glemieux if you're asking whether this new test could point to the same lightning file as any previously existing lightning test, then I agree. Your justification that the format of the stream files ultimately is the same, makes sense to me.

The specific ignitions file that Lara and I had used back when I worked on this happened to contain monthly data, but that need not be a defining characteristic of all ignitions files. I do not remember off the top of my head the temporal res. of the lightning stream, but maybe it's daily...
